starknet_py/net/account/account_client.py

```diff
@@ -307,6 +316,74 @@ async def sign_transaction(
 
         return execute_tx
 
+    async def sign_transaction(
+        self,
+        calls: Calls,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+        version: int = 0,
+    ) -> InvokeFunction:
+        """
+        Takes calls and creates signed InvokeFunction
+
+        :param calls: Single call or list of calls
+        :param max_fee: Max amount of Wei to be paid when executing transaction
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
+        :param version: Transaction version
+        :return: InvokeFunction created from the calls
+
+        .. deprecated:: 0.4.6
+            This method has been replaced by :meth:`AccountClient.create_invoke_transaction`
+        """
+        # TODO add correct deprecation version
+        warnings.warn(
+            "Sign_transaction is deprecated. Use create_invoke_transaction instead.",
+            category=DeprecationWarning,
+        )
+        return await self.create_invoke_transaction(
+            calls=calls, max_fee=max_fee, auto_estimate=auto_estimate, version=version
+        )
+
+    def create_declare_transaction(
+        self,
+        compilation_source: Optional[StarknetCompilationSource] = None,
+        compiled_contract: Optional[str] = None,
+        cairo_path: Optional[List[str]] = None,
+        max_fee: Optional[int] = None,
+        auto_estimate: bool = False,
+    ) -> Declare:
+        """
+        Create declaration tx.
+        Either `compilation_source` or `compiled_contract` is required.
+
+        :param compilation_source: string containing source code or a list of source files paths
+        :param compiled_contract: string containing compiled contract bytecode.
+                                  Useful for reading compiled contract from a file
+        :param cairo_path: a ``list`` of paths used by starknet_compile to resolve dependencies within contracts
+        :param max_fee: Max amount of Wei to be paid when executing transaction
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs
+        :return: A "Declare" transaction object
+        """
+        # pylint: disable=too-many-arguments
+        compiled_contract = create_compiled_contract(
+            compilation_source, compiled_contract, cairo_path
+        )
+        declare_tx = Declare(
+            contract_class=compiled_contract,
+            sender_address=self.address,
+            max_fee=0,
+            signature=[],
+            nonce=self._get_nonce(),
+            version=1,
```

Comment:
We should use version from account. Please fetch latest changes from 0.10 branch.